pkg/volume/glusterfs/glusterfs.go

```diff
@@ -288,23 +288,51 @@ func (c *glusterfsUnmounter) TearDownAt(dir string) error {
 
 func (b *glusterfsMounter) setUpAtInternal(dir string) error {
 	var errs error
-
 	options := []string{}
+	hasLogFile := false
+	log := ""
+
 	if b.readOnly {
 		options = append(options, "ro")
+
 	}
 
-	p := path.Join(b.glusterfs.plugin.host.GetPluginDir(glusterfsPluginName), b.glusterfs.volName)
-	if err := os.MkdirAll(p, 0750); err != nil {
-		return fmt.Errorf("failed to create directory %v: %v", p, err)
+	// Check logfile has been provided by user, if provided, use that as the log file.
+	for i, userOpt := range b.mountOptions {
+		if dstrings.Contains(userOpt, "log-file") {
+			glog.V(4).Infof("log-file mount option is provided")
+			hasLogFile = true
+			logOptions := dstrings.Split(userOpt, "=")
+
+			if len(logOptions) > 1 && logOptions[1] != "" {
```

Comment:
Do we need to check if user passed a file here? What does `mount.glusterfs` do when `log-options` has no `=` or is empty? Maybe the user does not want any log file at all.

I would suggest to simplify the logic to 

```go
for i, userOpt := range b.mountOptions {
    if dstrings.HasPrefix(userOpt, "log-file") {
        hasLogFile = true
        break
    }
}

if !hasLogFile {
        // use the default log file
}
```

This way, user is responsible to pass the right `log-file` value and validation is done in `mount.glusterfs`.

Reply:
I believe you meant `log-file` instead of `log-options` :), is yes, if user just pass `log-file` without `=` or `empty`, mount.glusterfs fails saying `unknown mountoption`, thats what I remember from my testing. I will reconfirm it once again.

Reply:
I think when user passes wrong option we should print some error or warning instead of silently falling back to something that the user definitely did not want.

Reply:
Sure, so it does a passthrough with latest patch set and if the option is wrong, it will fail.